src/controllers/users.controller.js

```diff
@@ -0,0 +1,78 @@
+const usersServices = require('../services/users.services');
+const { asyncHandler } = require('../utils/asyncHandler');
+const { getErrorWithStatus } = require('../utils/getError');
+
+const getUsers = async (_, res) => {
+  const users = await usersServices.getUsers();
+
+  res.status(200).json(users);
+};
+
+const getUserById = async (req, res) => {
+  const { id } = req.params;
+  const userId = Number(id);
+
+  if (isNaN(userId)) {
+    throw getErrorWithStatus(400, 'Invalid user ID');
+  }
+
+  const user = await usersServices.getUserById(userId);
+
+  if (!user) {
+    throw getErrorWithStatus(404, 'User not found');
+  }
+
+  res.status(200).json(user);
+};
+
+const createUser = async (req, res) => {
+  const { name } = req.body;
+
+  if (typeof name !== 'string') {
+    throw getErrorWithStatus(400, 'Wrong name field type,must be string');
```

Comment:
The error message has a minor typo: 'Wrong name field type,must be string'. Consider adding a space after the comma for better readability.